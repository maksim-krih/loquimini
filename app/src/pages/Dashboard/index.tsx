import { FC, useEffect, useState } from "react";
import { IProps } from "./types";
import { useStyles } from "./styles";
import { DashboardInfo } from "../../services/types";
import Api from "../../services";
import { Typography } from "antd";

const { Title } = Typography;

const Dashboard: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const [info, setInfo] = useState<DashboardInfo | undefined>(undefined);

  useEffect(() => {
    getDashboard();
  }, []);

  const getDashboard = () => {
    Api.Dashboard.getInfo()
    .then((response: DashboardInfo) => {
      setInfo(response);
    })
  };

  return (
    <div className={classes.container}>
      {info && 
        <>
          <Title level={4}>Total sum</Title>
          <div>{info.totalSum}</div>
          <br />
          <Title level={4}>Total debts</Title>
          <div>{info.totalDebts}</div>
          <br />
          <Title level={4}>Filled</Title>
          <div>{info.currentFilled}/{info.totalFilled}</div>
          <br />
          <Title level={4}>Paid</Title>
          <div>{info.currentPaid}/{info.totalPaid}</div>
          <br />
        </>
      }
    </div>
  );
}

export default Dashboard;