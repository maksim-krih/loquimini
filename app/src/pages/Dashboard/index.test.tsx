import Dashboard from ".";
import { shallow } from "enzyme";

const setUp = () => shallow(<Dashboard />);

describe("Dashboard", () => {
  it("should render Dashboard Component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
});