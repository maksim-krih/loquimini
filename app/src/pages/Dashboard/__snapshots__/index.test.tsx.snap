// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render Dashboard Component 1`] = `
<div
  className="container-0-2-1"
>
  <Title
    level={3}
  >
    July 2021
  </Title>
</div>
`;
